### ISSUE TITLE
🐛 `biome check --write` always counts Vue files as fixed, even when they did not change

### ISSUE THREAD

**Issue by moritzruth** [biomejs/biome#10666]
### Environment information

<details>

```bash

CLI:
  Version:                      2.5.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

```

</details>

### What happened?

1. Create a Vue file with a `<script>` element: `printf '<script>\n</script>' > test.vue`
2. Run `biome check --write --verbose test.vue`.
3. Run `biome check --write --verbose test.vue` again. As expected, the file will still not have changed.

Biome prints `Checked 1 file in 4ms. Fixed 1 file.` and the file is listed under "Files fixed", even though nothing changed.

Even if the file really was fixed by the first invocation, running it a second time should not change it again.

This does not happen when the file is completely empty or only contains the `<template>` section.

### Expected result

If Biome does not change the file, it should not count it as "fixed" in both the summary line and the verbose output.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by moritzruth** [comment]
Interestingly, this happens even with `--linter-enabled false --formatter-enabled false --assist-enabled false`.

**Comment by moritzruth** [comment]
@ematipico In #10515, the file actually changes. In my case, the file is byte-for-byte the same as before.

**Comment by ematipico** [comment]
It's still the same symptom.

**Comment by moritzruth** [comment]
By the same *symptom*, do you mean the same *cause*?

Sorry for being pedantic, I just want to make sure we're on the same page.

**Comment by ematipico** [comment]
I mean that the CLI writes the file on disk enough though it hasn't changed.

The piece of logic that is causing the two bugs is the same, I know because I worked on that part already

**Comment by moritzruth** [comment]
Alright, thanks!
